xmtp_mls/src/groups/disappearing_messages.rs

```diff
@@ -80,9 +91,22 @@ where
         Ok(())
     }
     async fn run(&mut self) -> Result<(), DisappearingMessagesCleanerError> {
-        if let Err(err) = self.delete_expired_messages().await {
-            tracing::error!("Error during deletion of expired messages: {:?}", err);
+        #[cfg(target_arch = "wasm32")]
+        let mut interval = Interval::new(INTERVAL_DURATION.as_millis() as u32);
+
+        #[cfg(not(target_arch = "wasm32"))]
+        let mut interval = time::interval(INTERVAL_DURATION);
+
+        loop {
+            #[cfg(target_arch = "wasm32")]
+            interval.next().await;
+
+            #[cfg(not(target_arch = "wasm32"))]
+            interval.tick().await;
+
+            if let Err(err) = self.delete_expired_messages().await {
```

Comment:
this error doesn't need to be printed here, we can use `?` and propagate it to the caller and handle it with the rest of the errors in `spawn_worker`.